byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -30,52 +31,53 @@
 
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private Key userKey;
-
-  @Override
-  public void init() {
-    Entity userEntity = new Entity("UserInputs");
-    userEntity.setProperty("start", "");
-    userEntity.setProperty("destinations", new ArrayList<String>());
-    datastore.put(userEntity);
-    userKey= userEntity.getKey();
-  }
-
+  private final UserService userService = UserServiceFactory.getUserService();
+  private Key tripKey;
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    tripKey = KeyFactory.stringToKey(request.getParameter("tripKey"));
+    addUserEntity(tripKey);
+
     Entity entity; 
     try{ 
-      entity = datastore.get(userKey);
+      entity = datastore.get(tripKey) ;
     } catch(EntityNotFoundException e){
-      UserLocations userLocations = new UserLocations("", new ArrayList<String>()); 
+      Trip trip = new Trip("","", new ArrayList<String>(), new ArrayList<String>(), new ArrayList<String>()); 
```

Comment:
If there is no trip entity, shouldn't we create a new one here and save it to datastore? Seems like this get request should handle creating new trip entities.

Reply:
the trip is now created in EntityServlet.java in the doPost. Eventually, there will be a button for users to click which calls doPost